mercury_agent/hardware/platform_detection.py

```diff
@@ -10,15 +13,18 @@ def is_platform(dmi_info, platform):
     return get_vendor(dmi_info) == platform
 
 
-def is_dell(dmi_info):
+def is_dell():
```

Comment:
These functions are not intended to be runtime dependent (requiring the global device_info datastructure to be populated). Revert these and update the callers.